scheduler/src/main/java/com/flipkart/flux/clientelb/dao/ClientElbDAOImpl.java

```diff
@@ -0,0 +1,135 @@
+ *  CRUD operations.
+ *
+ * @author akif.khan
+ */
+public class ClientElbDAOImpl implements ClientElbDAO {
+
+    private SessionFactoryContext sessionFactoryContext;
+
+    @Inject
+    public ClientElbDAOImpl(@Named("schedulerSessionFactoriesContext") SessionFactoryContext sessionFactoryContext) {
+        this.sessionFactoryContext = sessionFactoryContext;
+    }
+
+    /**
+     * Creates client elb entry in DB and returns saved object
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public ClientElb create(String clientElbId, ClientElb clientElb) {
+        return this.save(clientElb);
+    }
+
+    /**
+     * Retrieves Client Elb from DB by it's unique identifier
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+   public  ClientElb findById(String id) {
+        Criteria criteria = currentSession().createCriteria(ClientElb.class)
+                .add(Restrictions.eq("id", id));
+        Object object = criteria.uniqueResult();
+        ClientElb castedObject = null;
+        if(object != null)
+            castedObject = (ClientElb) object;
+        else
+            castedObject = null;
+        return castedObject;
+    }
+
+    /**
+     * Updates Elb Url of a client in DB
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public void updateElbUrl(String clientElbId, String clientElbUrl) {
+        Query updateQuery = currentSession().createQuery(
+                "update ClientElb set elbUrl = :clientElbUrl where id = :clientElbId");
+        updateQuery.setString("clientElbUrl", clientElbUrl != null ? clientElbUrl.toString() : null);
+        updateQuery.setString("clientElbId", clientElbId);
+        updateQuery.executeUpdate();
+    }
+
+    /**
+     * Delete Client Elb entry from DB
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public void delete(String id) {
+        Query deleteQuery = currentSession().createQuery("delete ClientElb c where c.id=:id");
+        deleteQuery.setString("id", id);
+        deleteQuery.executeUpdate();
+    }
+
+    /**
+     * Retrieve Older Client Elb entry limit by parameter rowCount
+     * @param rowCount
+     * @return List of oldest ClientElbs'
+     */
+    @Override
+    @Transactional
+    @SelectDataSource(storage = Storage.SCHEDULER)
+    public List<ClientElb> retrieveOldest(int rowCount) {
+        return currentSession()
+                .createCriteria(ClientElb.class)
+                .addOrder(Order.asc("createdAt"))
+                .setMaxResults(rowCount)
+                .list();
+    }
+
+    /**
+     * Saves the object in DB and returns the saved object.
+     * @param clientElb
+     * @return saved clientElb object
+     */
+    public ClientElb save(ClientElb clientElb) {
```

Comment:
Transactional?

Reply:
Removed this method and moved it's content to ONLY calling method which is Transactional:
     /**
     * Creates client elb entry in DB and returns saved object
     */
    @Override
    @Transactional
    @SelectDataSource(storage = Storage.SCHEDULER)
    public ClientElb create(String clientElbId, ClientElb clientElb) {
        return currentSession().save(clientElb);
    }